crates/core-subsystem/core-resolver/src/validation/validation_error.rs

```diff
@@ -74,6 +74,14 @@ pub enum ValidationError {
 
     #[error("Selection set too deep")]
     SelectionSetTooDeep(Pos),
+
+    #[error("Invalid value for '{value_name}': {range_detail}, {value_detail}")]
+    ValueOutOfRange {
+        value_name: String,
+        range_detail: String,
```

Comment:
May be not keep these as stings (instead keep the raw values) and let the formatting in the `error` macro take care.